reactioncommerce/reaction
# Refactor 3482 nnnnat replace imagemagick

Fixes #3482 
## Summary
Replaced GraphicMagick/ImageMagick with [sharp](http://sharp.pixelplumbing.com/en/stable/).
- Added sharp to the project and dynamically loading in the package.
- Updated image transforms to to use the `sharp()` functions.
- Refactored the `"Media" FS.Collection` to map the image transforms through a `buildGFS()` function to create each  `FS.Store.GridFS` collection.
- Updated and added new code comments.

## Testing
- Login as Admin and go to the Product Details Page.
- Open your network panel in dev-tools and select the `images` filter
- Upload a new product image and copy the new image's request URL from the network panel
- Open a new browser tab and paste the image's URL into the address bar
- At the very end of the URL after the token, you can add `&store={size}` to view each image transform.
- Image transforms sizes are `large`, `medium`, `small` & `thumbnail`
- Image transforms should be the same as the old `gm` crops

**Note to reviewer**
Wasn't sure if I'm doing the lazy load of sharp in the right way. I'm calling `lazyLoadSharp()` from within the `buildGFS()` function which is being mapped over when I create the `stores` Array. I'm guessing that the `if (sharp) return;` inside of `lazyLoadSharp()` is preventing sharp from being loaded more then once.

Never used JSDoc before so let me know if I'm failing at it
